airflow/operators/dagrun_operator.py

```diff
@@ -126,3 +144,22 @@ def execute(self, context: Dict):
                 dag.clear(start_date=self.execution_date, end_date=self.execution_date)
             else:
                 raise e
+
+        if self.wait_for_completion:
+            # wait for dag to complete
+            while True:
```

Comment:
I'm thinking if it's making sense to have something like `max_retries` or `timeout` here. Otherwise there is chance that this becomes a dead loop in extreme circumstances.

@turbaszek what do you think?

Reply:
Other than the idea/question above + my comments on the docstring, overall looks good to me.

Reply:
@XD-DENG I agree that's something we can do either here or in follow up PR fixing also the `ExternalTaskSensor` which also uses similar infinite loop. 

Personally I would be in favour of timeout. And probably something like `60 * 60 * 24` similar to what we have in `BaseSensor` to handle most of DAGs

Reply:
Sounds a good plan to me. Thanks @turbaszek

Reply:
Please also add an entry in `UPDATING.md` (section `master`) to describe this change @kukigai

Reply:
> Please also add an entry in `UPDATING.md` (section `master`) to describe this change @kukigai

@XD-DENG do we need it? The wait feature is a new addition and the default value is to avoid waiting, thus the default behaviour is the same as before this change.

Reply:
My concern was users don't get aware of new additions.

I don't have strong opinion on this though, so it's also ok for me if we don't add this entry in `UPDATING.md`.

Reply:
> My concern was users don't get aware of new additions.

This is a valid concern, however the UPDATING.md was purposed to be place for breaking changes. New features are "announced" in changelog 🤷‍♂️

Reply:
Yep. For no reason I mixed these two just now😂 Agree with you

Reply:
@turbaszek @XD-DENG 

regarding https://github.com/apache/airflow/pull/12126#discussion_r521673165, user can set execution_timeout.

Do you think we should have separate internal timeout for this operator?

I prefer to have user choose execution_timeout so keep infinite loop as is.

Reply:
> regarding #12126 (comment), user can set execution_timeout.

I agree, let's use the existing solution 👍